storage_controller/src/service.rs

```diff
@@ -218,9 +218,9 @@ fn passthrough_api_error(node: &Node, e: mgmt_api::Error) -> ApiError {
                 format!("{node} error receiving error body: {str}").into(),
             )
         }
-        mgmt_api::Error::ReceiveBody(str) => {
+        mgmt_api::Error::ReceiveBody(err) => {
             // Presume errors receiving body are connectivity/availability issues
-            ApiError::ResourceUnavailable(format!("{node} error receiving body: {str}").into())
+            ApiError::InternalServerError(anyhow::Error::from(err).context("error receiving body"))
```

Comment:
I think we can still get network errors here that should be 503s: our handling should be conditional on the `Kind` of the inner `reqwest` error -- there is a Kind::Decode that we can map to a 500 for serialization errors